Library/Homebrew/cleanup.rb

```diff
@@ -33,7 +35,15 @@ def self.cleanup_formula(formula)
     def self.cleanup_logs
       return unless HOMEBREW_LOGS.directory?
       HOMEBREW_LOGS.subdirs.each do |dir|
-        cleanup_path(dir) { dir.rmtree } if prune?(dir, :days_default => 14)
+        if Formulary.formula_ref? dir.basename.to_s
```

Comment:
We could switch it to detect formulae the other way, by excluding the names known to be installation steps ("install", "postinstall", "test") and assuming everything else is a formula.

Come to think of it, that may be more correct anyway, because formulae may have been removed or renamed over the course of a `brew update` since the installation was initially done.

It would accidentally move the logs from any external commands which are dropping their own log files under `HOMEBREW_LOGS`, but that seems like a small risk with low impact.